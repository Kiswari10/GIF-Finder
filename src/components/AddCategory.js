import React, { useState } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { IconContext } from "react-icons";
import { AiOutlineSearch } from "react-icons/ai";

const Form = styled.form`
  width: 100%;
  height: 55px;
  padding: 0;
  display: flex;
  flex-direction: row;
  input {
    width: 92%;
    height: 100%;
    outline: none;
    border: none;
    margin: 0;
    padding: 0 1%;
    font-size: 18px;
    font-weight: 600;
  }
  button {
    background-image: linear-gradient(45deg, #16bf69, #22c4dc 51%, #16bf69);
    height: 100%;
    width: 8%;
    min-width: 40px;
    border: none;
    outline: none;
    cursor: pointer;
    margin: 0;
    padding: 0;
  }
`;

Form.displayName = "form";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");
  const handleInputValue = (event) => {
    setInputValue(event.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((prev) => [inputValue, ...prev]);
    }
    setInputValue("");
  };
  return (
    <>
      <Form onSubmit={handleSubmit}>
        <input
          placeholder="Search GIFs"
          value={inputValue}
          onChange={handleInputValue}
        />
        <button onClick={handleSubmit}>
          <IconContext.Provider
            value={{
              size: "60%",
              color: "white",
              fontWeight: "bold",
            }}
          >
            <AiOutlineSearch />
          </IconContext.Provider>
        </button>
      </Form>
    </>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
