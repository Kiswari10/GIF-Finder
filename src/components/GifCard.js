import React from "react";
import { Card } from "react-bootstrap";

export const GifCard = ({ id, title, url }) => {
  return (
    <Card className="animate__animated animate__fadeIn">
      <Card.Img src={url} alt={title} />
    </Card>
  );
};
