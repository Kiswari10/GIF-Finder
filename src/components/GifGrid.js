import React, { useEffect, useState } from "react";
import { GifCard } from "./GifCard";
import { CardColumns } from "react-bootstrap";

export const GifGrid = ({ category }) => {
  const [gifs, setGifs] = useState([]);
  useEffect(() => {
    const getGifs = async () => {
      const response = await fetch(
        `http://api.giphy.com/v1/gifs/search?q=${category}&limit=12&api_key=PIJ9beNH3hdjavbtzJ5Nje2FdpV8cp0D`
      );
      const { data } = await response.json();
      const _gifs = data.map((img) => ({
        id: img.id,
        title: img.title,
        url: img.images.downsized_medium.url,
      }));
      setGifs(_gifs);
      console.log(_gifs);
    };
    getGifs();
  }, [category]);
  return (
    <div>
      <h3 className="logo">{category}</h3>
      <CardColumns>
        {gifs.map((gif) => (
          <GifCard {...gif} key={gif.id} />
        ))}
      </CardColumns>
    </div>
  );
};
