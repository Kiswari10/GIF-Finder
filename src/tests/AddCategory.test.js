import React from "react";
import "@testing-library/jest-dom";

import { shallow } from "enzyme";

import { AddCategory } from "../components/AddCategory";

describe("Pruebas en <AddCategory />", () => {
  const setCategories = jest.fn();
  let wraper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    wraper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("debe de retornar el componente correctamente", () => {
    expect(wraper).toMatchSnapshot();
  });

  test("debe cambiar el valor del input", () => {
    const input = wraper.find("input");
    const value = "Hello World";
    input.simulate("change", { target: { value } });
    const inputAfter = wraper.find("input");

    expect(inputAfter.props().value).toBe(value);
  });

  test("No debe enviar información con submit si el value del input esta vacío", () => {
    const form = wraper.find("form");
    form.simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });
});
