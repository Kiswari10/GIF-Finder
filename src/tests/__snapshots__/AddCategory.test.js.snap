// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe de retornar el componente correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Search GIFs"
      value=""
    />
    <button
      onClick={[Function]}
    >
      <ContextProvider
        value={
          Object {
            "color": "white",
            "fontWeight": "bold",
            "size": "60%",
          }
        }
      >
        <AiOutlineSearch />
      </ContextProvider>
    </button>
  </form>
</Fragment>
`;
